### PR TITLE
add noderesource controller

### PR DESCRIPTION
**Pull request by jinze1107**

### CODE REVIEW

File: pkg/controller/noderesource/node_resource_controller.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+	// 4.update Node status extend-resource info
+	// https://kubernetes.io/zh/docs/tasks/administer-cluster/extended-resource-node/
+	// TODO fix: strategic merge patch kubernetes
+	if err := r.Client.Status().Update(context.TODO(), nodeCopy); err != nil {
+		r.Recorder.Event(node, v1.EventTypeNormal, "FailedUpdateNodeExtendResource", err.Error())
+		r.Log.Error(err, "update Node status extend-resource error: %v", err)
+		return ctrl.Result{}, err
+	}
+	r.Recorder.Event(nodeCopy, v1.EventTypeNormal, "UpdateNode", "Update Node Extend Resource Success")
```
Comment (mfanjie) [inline pkg/controller/noderesource/node_resource_controller.go:108]: please add updated resource name/value to this event

File: pkg/controller/noderesource/node_resource_controller.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+// the NodeResource object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.10.0/pkg/reconcile
+func (r *NodeResourceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	r.Log.Info("node resource reconcile", "node-resource", req.NamespacedName, req.Name)
+
+	nodePre := &predictionapi.NodePrediction{}
```
Comment (mfanjie) [inline pkg/controller/noderesource/node_resource_controller.go:56]: please integrated with tsp, the new api is ready

File: pkg/controller/noderesource/node_resource_controller.go [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+			}
+
+			if key == string(v1.ResourceMemory) {
+				nodeCopy.Status.Capacity["ext-resource.node.crane.io/memory"] = *resource.NewQuantity(resourceValue, resource.DecimalSI)
+			}
+		}
+	}
+
+	// 4.update Node status extend-resource info
+	// https://kubernetes.io/zh/docs/tasks/administer-cluster/extended-resource-node/
```
Comment (wenlxie) [inline pkg/controller/noderesource/node_resource_controller.go:101]: Can it be changed to https://kubernetes.io/docs/tasks/administer-cluster/extended-resource-node/ ?

**Comment by mfanjie** [comment]
commit is merged in different PR, close this one.

**Pull request closed without merging**